plugins/store/ios/Sources/StorePlugin.swift

```diff
@@ -0,0 +1,217 @@
+// Copyright 2019-2023 Tauri Programme within The Commons Conservancy
+// SPDX-License-Identifier: Apache-2.0
+// SPDX-License-Identifier: MIT
+
+import Foundation
+
+import SwiftRs
+import Tauri
+import UIKit
+import WebKit
+
+
+struct SaveStore: Codable {
+    let store: String
+    let cache: [String: JSON]
+}
+
+class StorePlugin: Plugin {
+    @objc public func save(_ invoke: Invoke) throws {
+        do {
+            let args = try invoke.parseArgs(SaveStore.self)
+            let store = args.store
+            let cache = args.cache
+            let fileURL = getUrlFromPath(path: store)
+            
+            try JSONEncoder().encode(cache).write(to: fileURL)
+            invoke.resolve()
+        } catch {
+            invoke.reject(error.localizedDescription)
+        }
+    }
+    
+    @objc public func load(_ invoke: Invoke) throws {
+        do {
+            let path = try invoke.parseArgs(String.self)
+            let fileURL = getUrlFromPath(path: path)
+            let data = try String(contentsOf: fileURL)
+            let passData = dictionary(text: data)
+            
+            invoke.resolve(passData)
+        } catch {
+            invoke.reject(error.localizedDescription)
+        }
+    }
+    
+    func dictionary(text: String) -> [String: Any?] {
+        if let data = text.data(using: .utf8) {
+            do {
+                return try JSONSerialization.jsonObject(with: data, options: []) as! [String: Any]
+            } catch {
+                fatalError(error.localizedDescription)
+            }
+        }
+        
+        return [:]
+    }
+    
+    func getUrlFromPath(path: String) -> URL {
+        do {
+            var url = try FileManager.default
+                .url(
+                    for: .applicationSupportDirectory,
+                    in: .userDomainMask,
+                    appropriateFor: nil,
+                    create: true
+                )
+            let components = path.split(separator: "/").map { element in String(element) }
+            
+            if components.count == 1 {
+                return url.appendPath(path: path)
+            }
+            
+            for i in 0..<components.count {
+                url = url.appendPath(path: components[i])
+            }
+            
+            if components.count > 1 {
+                try FileManager.default.createDirectory(at: url, withIntermediateDirectories: true)
+            }
+            
+            url = url.appendPath(path: components.last!)
+            
+            return url
```

Comment:
can't this be simplified to

```suggestion
            let components = path.split(separator: "/").map { element in String(element) }
            
            for i in 0..<components.count {
                if i == components.count - 1 {
                    try FileManager.default.createDirectory(at: url, withIntermediateDirectories: true)
                }
            
                url = url.appendPath(path: components[i])
            }
            
            return url
```